### DISCUSSION TITLE
Get feature SourceId

### DISCUSSION THREAD

**Discussion by levelost** [openlayers/ol-mapbox-style#1159] [Q&A] [answered]
I want to display information about an object when clicked. I do myMapboxVectorLayer.getFeatures(e.pixel). I get the feature. But I don’t know from what source this feature comes. How can I find out the SourceId of a feature?

**Comment by ahocevar** [comment]
See the `recordStyleLayer()` function. It will store the Mapbox layer on the feature, from which you can get the `source`.

**Reply by levelost** [reply]
I discovered that «layer» is already written in the properties. Is it possible to write «mapbox-layer» and not write «layer»?

**Comment by ahocevar** [comment, marked as answer]
`layer` comes from the vector tiles, and is set by ol/format/MVT. It has a different meaning - that‘s the `source-layer` in the Mapbox Style‘s `layer`.
